### DISCUSSION TITLE
Need Help for Generating and uploading JTL file with run locust more than 1 workers

### DISCUSSION THREAD

**Discussion by RifkiNurgraha** [ludeknovy/jtl-reporter#72] [Q&A] [answered]
i use jtl reporter for locust framework. but if i use workers more than one. csv log only create one not mix from all workers. any insight about this ? thank you

**Comment by ludeknovy** [comment, marked as answer]
Hi @RifkiNurgraha what version of locust do you use and what listener do you use?

**Reply by RifkiNurgraha** [reply]
Hi, im using locust 1.5.3, and jtl_listener.py

**Reply by ludeknovy** [reply]
@RifkiNurgraha I checked the issue and I was able to reproduce it in case I did not provide the `JTL_API_TOKEN`, and that resulted in `KeyError: 'JTL_API_TOKEN'` error. Once I added it, it seems to be working correctly. 

Were you getting the `KeyError: 'JTL_API_TOKEN'`  error when experiencing this issue?

**Reply by RifkiNurgraha** [reply]
i dont get that issue. for jtl api token i think just open the dashboard jtl ( http://localhost:2020/dashboard) and go to  admin -> administrate ->api keys -> add token, and then set the token to jtl_listener.py, right ?

**Reply by ludeknovy** [reply]
@RifkiNurgraha Yes, thats correct.

I tried it again(locust 1.5.3, two workers) and locust reports N samples. The same number I find in CSV and in the application itself. I tried it with this [listener](https://github.com/ludeknovy/jtl-reporter/blob/master/scripts/jtl_listener.py).

You could check whether both workers do report something to master, by adding debug logs in `_worker_report` method.

**Reply by RifkiNurgraha** [reply]
hm i still got the same issue 😂. csv log only create one not mix from all workers, actually im using this cmd for example `locust -f test/test_user2.py --expect-workers 3 --master` and hit this 3 times in terminal `locust -f test/test_user2.py --worker --master-host=localhost` . and in my file.py ->

@events.init.add_listener
def on_locust_init(environment, **_kwargs):
    JtlListener(env=environment, project_name="locust",
                scenario_name="User",
                environment="staging")

can i see example of your file.py ?

**Reply by RifkiNurgraha** [reply]
wanna see my repository github ? and help me to check it direcly 😂

**Reply by ludeknovy** [reply]
@RifkiNurgraha yes, please share your github repository!

**Reply by RifkiNurgraha** [reply]
sure, this is the repository https://github.com/RifkiNurgraha/locust-pt, for example test at test/test_user2.py. thank you

**Reply by ludeknovy** [reply]
Run it using your repository (`test_user2.py`) and it still looks correct to me:

![Snímek obrazovky 2021-05-24 v 15 22 20](https://user-images.githubusercontent.com/13610612/119354515-75168280-bca4-11eb-9572-f54dd3fc16d6.png)

The CSV has +1 line because of the header.

I ran it with 3 workers.

Btw. in the repository you dont use `-u` flag with number of users. It defaults to 1. It means only 1 worker will generate the load, the others will be idle.

**Reply by RifkiNurgraha** [reply]
sorry, I think we misunderstood 😂 . let me explain about the issue again. 

- first i run my script  (test_user2.py) with 3 workers.
![image](https://user-images.githubusercontent.com/39902589/119358720-19172280-bcd3-11eb-8244-703f3e515809.png)
as you can see at the picture, i swarm 100 user with have 3 workers each produce 33 users. 
- second i stop the test. and automatically upload to jtl dasboard , using jtl_listener.py and upload_jtl.py at 'locust-pt/libs/'
![image](https://user-images.githubusercontent.com/39902589/119359262-a3f81d00-bcd3-11eb-9a53-0ab9b34c21a5.png)
but as you can see in the dashboard the result of total user only 34, which is the correct is 100 ( total from 3 workers), right ?

any solution regarding this ?

**Reply by ludeknovy** [reply]
Ahh, ok. I see the issue now.

You need to provide "Hostname" column. 

Add  `self._worker_id = random.randint(10000,9999999)` to the JtlListener constructor.

Then add `Hostname` to the `csv_headers`. And in `add_result` method add `Hostname = str(self._worker_id)` and in the same method put `Hostname` into `row`.

After doing this It showed me 100 VU in the app.

I will prepare fix for it later. Thanks for raising this issue.

**Reply by RifkiNurgraha** [reply]
Thanks a bunch for your fast response. you nailed it. 🥳

**Reply by ludeknovy** [reply]
@RifkiNurgraha could you please mark it as answered, please? 

Thanks again for reporting it!
